extensions/ql-vscode/src/vscode-tests/cli-integration/remote-queries/variant-analysis-manager.test.ts

```diff
@@ -149,6 +149,41 @@ describe('Variant Analysis Manager', async function() {
 
         expect(getVariantAnalysisRepoResultStub.calledOnce).to.be.true;
       });
+
+      it('should pop download tasks off the queue', async () => {
+        const getResultsSpy = sandbox.spy(variantAnalysisManager, 'autoDownloadVariantAnalysisResult');
+
+        const input = [
+          variantAnalysisManager.queue.add(async () => {
+            await variantAnalysisManager.autoDownloadVariantAnalysisResult(
+              scannedRepos[0],
+              variantAnalysis,
+              cancellationTokenSource.token
+            );
+          }),
```

Comment:
You shouldn't need to create a new `async` function which awaits the call, it should be possible to skip the `await` since the `autoDownloadVariantAnalysisResult` creates a promise anyway:

```suggestion
          variantAnalysisManager.queue.add(variantAnalysisManager.autoDownloadVariantAnalysisResult(
            scannedRepos[0],
            variantAnalysis,
            cancellationTokenSource.token
          )),
```

Reply:
I did try this, but the `.add()` method expects a function rather than a promise. 

![Screenshot 2022-10-17 at 14 39 30](https://user-images.githubusercontent.com/1354439/196212307-da48d8b3-a7c0-4e43-a876-3a838eec2949.png)